django/db/models/base.py

```diff
@@ -742,9 +743,9 @@ def refresh_from_db(self, using=None, fields=None, from_queryset=None):
                     field.delete_cached_value(self)
 
         # Clear cached relations.
-        for field in self._meta.related_objects:
-            if (fields is None or field.name in fields) and field.is_cached(self):
-                field.delete_cached_value(self)
+        for rel in self._meta.related_objects:
+            if fields is None and rel.is_cached(self):
```

Comment:
The `field.name in fields` clause would never match, because rel objects aren't fields. It can be removed in favour of only clearing relations if an explicit set of field name wasn't provided.

Reply:
I think it would match under some circumstances when a related field was explicitly requested to be refreshed 

https://github.com/django/django/blob/1722f2db5808708de6fc6e0f48af2d518be1e348/tests/basic/tests.py#L1094-L1099

Reply:
This was wrong, as evidenced by the test failure: `basic.tests.ModelRefreshTests.test_refresh_clears_reverse_related_explicit_fields`.

@g-nie added this in the previous commit, so nice catch!